lib/blocktree/blocktree_test.go

```diff
@@ -386,12 +411,46 @@ func TestBlockTree_Prune(t *testing.T) {
 	}
 }
 
+func TestBlockTree_PruneCache(t *testing.T) {
+	var bt *BlockTree
+	var branches []testBranch
+
+	for {
+		bt, branches = createTestBlockTree(testHeader, 5, nil)
+		if len(branches) > 0 && len(bt.getNode(branches[0].hash).children) > 1 {
+			break
+		}
+	}
+
+	// pick some block to finalise
+	finalised := bt.head.children[0].children[0].children[0]
+	pruned := bt.Prune(finalised.hash)
+
+	for _, prunedHash := range pruned {
+		block, ok := bt.nodeCache[prunedHash]
+
+		require.False(t, ok)
+		require.Nil(t, block)
+	}
+
+}
+
 func TestBlockTree_DeepCopy(t *testing.T) {
 	bt, _ := createFlatTree(t, 8)
 
 	btCopy := bt.DeepCopy()
 
 	require.Equal(t, bt.db, btCopy.db)
+	for hash := range bt.nodeCache {
+		b, ok := btCopy.nodeCache[hash]
+		b2 := bt.nodeCache[hash]
+
+		require.True(t, ok)
+		require.True(t, b != b2)
+
+		require.True(t, equalNodeValue(b, b2))
```

Comment:
please test deep copy by checking that modifying a node with some hash in the copy doesn't modify the node with the same hash in the original